tests/link/pytorch/test_basic.py

```diff
@@ -72,104 +72,116 @@ def compare_pytorch_and_py(
+                self.called = 0
 
-        def make_node(self, *args):
-            return Apply(self, list(args), [x.type() for x in args])
+            def make_node(self, *args):
+                return Apply(self, list(args), [x.type() for x in args])
 
-        def perform(self, inputs, outputs):
-            for i, inp in enumerate(inputs):
-                outputs[i][0] = inp[0]
+            def perform(self, inputs, outputs):
+                for i, inp in enumerate(inputs):
+                    outputs[i][0] = inp[0]
 
-    @pytorch_funcify.register(TestOp)
-    def pytorch_funcify_TestOp(op, **kwargs):
-        def func(*args, op=op):
-            op.called += 1
-            return list(args)
+        @pytorch_funcify.register(TestOp)
+        def pytorch_funcify_TestOp(op, **kwargs):
+            def func(*args, op=op):
+                op.called += 1
+                return list(args)
 
-        return func
+            return func
 
-    op1 = TestOp()
-    op2 = TestOp()
+        op1 = TestOp()
+        op2 = TestOp()
 
-    q, r = op1(x, y)
-    outs = op2(q + r, q + r)
+        q, r = op1(x, y)
+        outs = op2(q + r, q + r)
 
-    out_fg = FunctionGraph([x, y], outs, clone=False)
-    assert len(out_fg.outputs) == 2
+        out_fg = FunctionGraph([x, y], outs, clone=False)
+        assert len(out_fg.outputs) == 2
 
-    out_torch = pytorch_funcify(out_fg)
+        out_torch = pytorch_funcify(out_fg)
 
-    x_val = torch.tensor([1, 2]).to(getattr(torch, config.floatX))
-    y_val = torch.tensor([2, 3]).to(getattr(torch, config.floatX))
+        x_val = torch.tensor([1, 2]).to(getattr(torch, config.floatX))
+        y_val = torch.tensor([2, 3]).to(getattr(torch, config.floatX))
 
-    res = out_torch(x_val, y_val)
-    assert len(res) == 2
-    assert op1.called == 1
-    assert op2.called == 1
+        res = out_torch(x_val, y_val)
+        assert len(res) == 2
+        assert op1.called == 1
+        assert op2.called == 1
 
-    res = out_torch(x_val, y_val)
-    assert len(res) == 2
-    assert op1.called == 2
-    assert op2.called == 2
+        res = out_torch(x_val, y_val)
+        assert len(res) == 2
+        assert op1.called == 2
+        assert op2.called == 2
 
 
-def test_shared():
-    a = shared(np.array([1, 2, 3], dtype=config.floatX))
-    pytensor_torch_fn = function([], a, mode="PYTORCH")
-    pytorch_res = pytensor_torch_fn()
+@pytest.mark.parametrize("device", ["cpu", "cuda"])
+def test_shared(device):
+    with torch.device(device):
+        a = shared(np.array([1, 2, 3], dtype=config.floatX))
+        pytensor_torch_fn = function([], a, mode="PYTORCH")
+        pytorch_res = pytensor_torch_fn()
 
-    assert isinstance(pytorch_res, torch.Tensor)
-    np.testing.assert_allclose(pytorch_res.cpu(), a.get_value())
+        assert isinstance(pytorch_res, torch.Tensor)
+        assert isinstance(a.get_value(), np.ndarray)
+        np.testing.assert_allclose(pytorch_res.cpu(), a.get_value())
```

Comment:
Can you assert `a.get_value()` returns a `np.ndarray`?

Reply:
This must always be a numpy ndarray. You may be write about non shared outputs being Tensors. I think they are JAX arrays in the JAX backend so it's fine if those are tensors